### PR TITLE
[Issue #193] Redid my work after wrong merge into master

### PR DESCRIPTION
**Pull request by m0hamed**
Issue #193

### CODE REVIEW

File: arability/config/locales/en.yml [hunk 1]
```diff
@@ -70,3 +70,5 @@ en:
   stats_age: "Age Statistics"
   stats_education: "Education Statistics"
   stats_no_votes: "Sorry! this synonym has no votes yet!"
+
```
Comment (NourhanMohamed) [inline arability/config/locales/en.yml]: remove the extra 2 new lines

File: arability/config/locales/ar_extra.yml [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+    abbr_day_names:
+    - الأحد
+    - الإثنين
+    - الثلاثاء
+    - الأربعاء
+    - الخميس
+    - الجمعة
+    - السبت
+    abbr_month_names:
+    -
```
Comment (NourhanMohamed) [inline arability/config/locales/ar_extra.yml:12]: Is that empty on purpose?

File: arability/config/locales/ar_extra.yml [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+    - الأربعاء
+    - الخميس
+    - الجمعة
+    - السبت
+    formats:
+      default: ! '%Y-%m-%d'
+      long: ! '%B %e, %Y'
+      short: ! '%e %b'
+    month_names:
+    -
```
Comment (NourhanMohamed) [inline arability/config/locales/ar_extra.yml:38]: same here?

File: arability/config/locales/ar_extra.yml [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+      hour: ساعة
+      minute: دقيقة
+      month: الشهر
+      second: ثانية
+      year: السنة
+  errors: &errors
+    format: ! '%{attribute} %{message}'
+    messages:
+      accepted: يجب أن تقبل
+      blank: فارغ، يرجى ملء الحقل
```
Comment (NourhanMohamed) [inline arability/config/locales/ar_extra.yml:146]: الحقل!! Guys did u just grab a file from over the internet and pasted it?

Comment (m0hamed) [inline arability/config/locales/ar_extra.yml:146]: yes, yes and yes

File: arability/app/controllers/application_controller.rb [hunk 5]
```diff
@@ -1,8 +1,43 @@
...
+  # Author:
+  #   Mohamed Ashraf
+  # Desciption:
+  #   This function sets the locale to the default locale of ar or the
+  #   whichever locale stored in the session. If a locale is chosen it is
+  #   automatically stored in the session.
+  # params:
+  #   locale: from the url if exists
+  # success:
+  #   --
```
Comment (mostafahassaan) [inline arability/app/controllers/application_controller.rb]: explain success, be descriptive.

File: arability/app/controllers/application_controller.rb [hunk 6]
```diff
@@ -1,8 +1,43 @@
...
+  end
+
+  # Author:
+  #   Mohamed Ashraf
+  # Description:
+  #   It adds the current locale to the url if not specified
+  # params:
+  #   locale: from the url if exists
+  # success:
+  #   --
```
Comment (mostafahassaan) [inline arability/app/controllers/application_controller.rb]: add to success that it sets url

**Comment by mostafahassaan** [comment]
@m0hamed small issue with docs, checkout comments on file

File: arability/app/controllers/application_controller.rb [hunk 7]
```diff
@@ -1,8 +1,43 @@
...
+  #   automatically stored in the session.
+  # params:
+  #   locale: from the url if exists
+  # success:
+  #   --
+  # failure:
+  #   --
+  def set_locale
+    if params[:locale].nil?
+      if session[:locale].nil?
```
Comment (NourhanMohamed) [inline arability/app/controllers/application_controller.rb]: use I18n.locale = :ar if session[:locale].nil?

same goes for english

File: arability/app/controllers/application_controller.rb [hunk 8]
```diff
@@ -1,8 +1,43 @@
...
+  #   Mohamed Ashraf
+  # Description:
+  #   It adds the current locale to the url if not specified
+  # params:
+  #   locale: from the url if exists
+  # success:
+  #   --
+  # failure:
+  #   --
   def default_url_options(options={})
```
Comment (NourhanMohamed) [inline arability/app/controllers/application_controller.rb:36]: space before and after the equal of the optional parameter

File: arability/config/locales/ar_extra.yml (hunk 2 above)
Comment (m0hamed) [inline arability/config/locales/ar_extra.yml:12]: yes

File: arability/config/locales/ar_extra.yml (hunk 3 above)
Comment (m0hamed) [inline arability/config/locales/ar_extra.yml:38]: yes

**Pull request merged**